File: src/database/models/game.js
const mongoose = require("mongoose");

const Game = new mongoose.Schema({
    guildID: { type: String, index: true },
    channelID: { type: String, index: true },
    createdBy: { type: String },
    wonBy: { type: String },
    answer: { type: Number },
    points: { type: Number, default: 0 },
    guesses: { type: Number, default: 0 },
    finishedAt: { type: Date, index: true },
    finished: { type: Boolean, default: false },
    disabled: { type: Boolean, default: false },
    createdAt: { type: Date, default: Date.now },
    updatedAt: { type: Date, default: Date.now },
});

module.exports = mongoose.model("game", Game);

File: internal/domain/game.go
package domain

import (
	"context"
	"time"
)

type Game struct {
	ID                string    `bson:"_id,omitempty"`
	GuildID           string    `bson:"guildID"`
	ChannelID         string    `bson:"channelID"`
	WinMessageID      string    `bson:"winMessageID"`
	WonBy             string    `bson:"wonBy"`
	Answer            int64     `bson:"answer"`
	Points            int64     `bson:"points"`
	Guesses           int64     `bson:"guesses"`
	FinishedAt        time.Time `bson:"finishedAt"`
	Finished          bool      `bson:"finished"`
	Disabled          bool      `bson:"disabled"`
	CreatedBy         string    `bson:"createdBy"`
	CreatedAt         time.Time `bson:"createdAt"`
	UpdatedAt         time.Time `bson:"updatedAt"`
	AutoReactionHints bool      `bson:"autoReactionHints"`
}

type GameRepository interface {
	Create(ctx context.Context, game *Game) error
	Finish(ctx context.Context, gameID, messageID, wonBy string, guesses int64) error
	GetRunningInChannel(ctx context.Context, channelID string) (*Game, error)
	IncrementGuesses(ctx context.Context, gameID, channelID string) (int64, error)
	GetDailyLeaderboard(ctx context.Context, guildID string) ([]Game, error)
	GetAllTimeLeaderboard(ctx context.Context, guildID string) ([]Game, error)
	GetGamesBetweenDates(ctx context.Context, guildID string, startDate, endDate time.Time) ([]Game, error)
}
